app/instrumentation-tests/src/org/commcare/androidTests/DateWidgetsTests.kt

```diff
@@ -0,0 +1,208 @@
+        InstrumentationUtility.nextPage()
+        InstrumentationUtility.nextPage()
+        for (i in 1..12){
+            val month_text = InstrumentationUtility.getText(onView(withId(R.id.monthtxt)))
+            if (month_text != null) {
+                assertTrue(listOfEthiopianMonths.contains(month_text))
+            }
+            onView(withId(R.id.monthupbtn)).perform(ViewActions.click())
+        }
+
+        onView(withId(R.id.yeardownbtn)).perform(ViewActions.click())
+        val date_selected = setDateToUniversalCalender(-10,0,0)
+        val gregorian_date = InstrumentationUtility.getText(onView(withId(R.id.dateGregorian)))
+        val formatted_date = formatGregorianDate(gregorian_date.drop(1).dropLast(1))
+
+        InstrumentationUtility.nextPage()
+        Log.i("Ethiopian date:", date_selected)
+        Log.i("Gregorian date:", formatted_date)
+        assertTrue(onView(withSubstring(formatted_date)).isPresent())
+        assertTrue(onView(withSubstring(date_selected)).isPresent())
+        InstrumentationUtility.submitForm()
+        assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+    }
+
+    @Test
+    fun testStandardWidget(){
+
+        InstrumentationUtility.openModule(4)
+        InstrumentationUtility.rotateLeft()
+        val dateList = setDateTo(-10)
+//        Thread.sleep(1000)
+        InstrumentationUtility.rotatePortrait()
+        InstrumentationUtility.nextPage()
+
+        for (formats in dateList){
+            assertTrue(onView(allOf(withSubstring(formats))).isPresent())
+        }
+
+        InstrumentationUtility.submitForm()
+        assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+    }
+
+    /**
+     * function to set date in the date picker and return different format of the selected date
+     */
+    fun setDateTo(days: Int): Array<String> {
+        val calendar = Calendar.getInstance()
+        calendar.add(Calendar.DATE, +days)
+
+        val year: Int = calendar.get(Calendar.YEAR)
+        val month: Int = calendar.get(Calendar.MONTH) + 1
+        val day: Int = calendar.get(Calendar.DAY_OF_MONTH)
+        onView(
+            ViewMatchers.withClassName(
+                Matchers.equalTo(
+                    DatePicker::class.java.name
+                )
+            )
+        ).perform(PickerActions.setDate(year, month, day))
+
+        val long_format = SimpleDateFormat("EE, MMM dd, yyyy").format(calendar.time)
+        val short_format = SimpleDateFormat("d/M/yy").format(calendar.time)
+        val unformatted_date = SimpleDateFormat("yyyy-MM-dd").format(calendar.time)
+        return arrayOf(long_format.toString(),short_format.toString(),unformatted_date.toString())
+    }
+
+    /**
+     * function to set date in the date universal picker
+     */
+    fun setDateToUniversalCalender(days: Int, months: Int, years: Int) : String{
```

Comment:
Is there a reason we can't just do `PickerActions.setDate(year, month, day)` like in `setDateTo` method ?